docs/howto/override-project-templates.txt

```diff
@@ -0,0 +1,248 @@
+==============================================================
+How to override project and app file structure using templates
+==============================================================
+
+
+Introduction
+============
+
+The first command you will run when setting up a django project is
+:djadmin:`startproject`. Having set up your project, you will then start
+creating your apps with :djadmin:`startapp`. Each of these commands will create
+a file structure that is based on default templates within the Django
+repository.
+
+However, you can create your own templates so that when you run
+:djadmin:`startproject` or :djadmin:`startapp` it creates the app and project
+structure that is best for you.
+
+What the defaults look like
+===========================
+
+The command :djadmin:`startproject` will create the ``manage.py`` file, along
+with a folder named after the ``project_name`` parameter passed in the initial
+command. Here the command is run with the project name as ``myproject``:
+
+.. console::
+
+   $ django-admin startproject myproject .
+
+The resulting file structure is based on the default template-file structure in
+(:source:`django/conf/project_templates`). There, each file has the extension
+``.py-tpl`` to denote the file as a template. Each template contains template
+tags that end up being filled from the command and django version context. The
+template context is detailed at the end of the documentation for
+:djadmin:`startproject`.
+
+The resulting files all have the ``.py`` file extension and are ready to be
+used:
+
+.. code-block:: none
+
+    myproject/
+        manage.py
+        myproject/
+            __init__.py
+            asgi.py
+            settings.py
+            urls.py
+            wsgi.py
+
+A similar thing happens when you run :djadmin:`startapp`. Both commands inherit
+from the same ``TemplateCommand`` class (in
+:source:`django/core/management/templates.py`), but they discern between app
+and project.
+
+A typical command to initialize an app with the name ``myapp``:
+
+.. console::
+
+   $ python manage.py startapp myapp .
+
+This creates an app in the current directory with the default file structure
+found in (:source:`django/conf/app_templates`). As above, each template  has
+the extension ``.py-tpl`` to denote the file as a template, but is copied over
+as a python file. Each template contains template tags for the variables that
+will be passed.
+
+.. code-block:: none
+
+    myapp/
+        __init__.py
+        admin.py
+        apps.py
+        migrations/
+            __init__.py
+        models.py
+        tests.py
+        views.py
+
+
+Creating your own Project and App Templates
+===========================================
+
+The default project and app file structures are the ones suggested by django,
+but not the only ones that can be used. Because django does not require a
```

Comment:
```suggestion
The default project and app file structures are the ones suggested by Django,
but not the only ones that can be used. Because Django does not require a
```